.github/workflows/nox-session.yml

```diff
@@ -0,0 +1,148 @@
+name: Run nox session
+run-name: "[${{ github.ref_name }}] Run ${{ inputs.nox-session }} - python ${{ inputs.python-version }} - ${{ inputs.os }}"
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        type: string
+        required: true
+      os:
+        type: string
+        required: true
+      nox-session:
+        type: string
+        required: true
+      generate-coverage-report:
+        type: boolean
+        default: false
+      pytket-use-latest-pre-release:
+        type: boolean
+        default: false
+
+  workflow_dispatch:
+    inputs:
+      python-version:
+        description: "Python version to use"
+        type: choice
+        default: "3.10"
+        required: true
+        options:
+          - "3.10"
+          - "3.11"
+      os:
+        description: "Runner OS"
+        type: choice
+        default: "ubuntu-22.04"
+        required: true
+        options:
+          - "ubuntu-22.04"
+          - "macos-12"
+          - "windows-2022"
+      nox-session:
+        description: "Nox session to run"
+        type: choice
+        required: true
+        options:
+          - tests
+          - mypy
+          - pre-commit
+          - docs-build
+      generate-coverage-report:
+        description: Generate coverage report (for tests session)
+        type: boolean
+        default: false
+      pytket-use-latest-pre-release:
+        description: Update pytket to latest pre-release
+        type: boolean
+        default: false
```

Comment:
Can we set this to true?

Reply:
We could. For me it seems more logical to have it as `false`. My logic is that the extension should definitely work with the latest stable `pytket`, because that's what users will be using it with most likely. Making sure it works with pre-releases is good, since it's more likely it will work with the next stable release, but seems weird to me to have that a the default. Either way it doesn't really matter, since we check both. So if you think it's better as default true we can change it.

Reply:
I think it would be better to change the default, because main reason we create a pre release is if we want to check that all the extensions are still working with the latest changes